dlt/extract/resource.py

```diff
@@ -343,6 +343,19 @@ def _gen_wrap(gen: TPipeStep) -> TPipeStep:
             self._pipe.replace_gen(_gen_wrap(self._pipe.gen))
         return self
 
+    def parallelize(self) -> "DltResource":
+        """Wraps the resource to execute each item in a threadpool to allow multiple resources to extract in parallel."""
+
+        if (
+            not inspect.isgenerator(self._pipe.gen)
```

Comment:
does that really work? IMO you cannot parallelize executing generator. you need to write a test for that. start generator in the main thread and then use parallelize and evaluate it. it must fail. we should detect and raise a proper exception.

Reply:
This works fine, or am I missing something? See: https://github.com/dlt-hub/dlt/blob/sthor%2Fparallelize-decorator/tests/pipeline/test_resources_evaluation.py#L391-L414

It's still executing the generator in one thread at a time, the `while busy: yield None` loop acts as a lock